### ISSUE TITLE
Can't navigate back from ADFS login portal

### ISSUE THREAD

**Issue by candu** [CityofToronto/bdit_flashcrow#674]
**What Happened**
A logged-out user clicked Track Requests and was presented with the City login portal, but decided not to proceed and instead clicked the Back button.  Instead of returning to the map, they were redirected back to the City login portal.

**What Should Happen**
If the user cancels out of the City login portal, or if they go back, they should not be redirected back into the City login portal.

**To Reproduce**
Steps to reproduce the bug:
1. Ensure you are logged out.
2. Click on "Track Requests" in the left navbar.
3. Instead of logging in, either go back or click "Cancel".

At this point, instead of returning to the map, you are redirected back to the login portal.

**Additional Notes**
This looks to be an issue with our `router.beforeEach` hook, which [calls `restoreLoginState`](https://github.com/CityofToronto/bdit_flashcrow/blob/master/web/router.js#L260) before doing anything else.

Inside `restoreLoginState`, we look for an entry in `window.sessionStorage` containing the path the user was attempting to load before logging in.  If this entry exists, we redirect them there.  However, this doesn't capture the case where the user doesn't complete the login flow - in which case the user is redirected, but they haven't logged in, so they're redirected to do so.

In other words: once a user clicks an action that requires login, right now they either have to complete the login process _or_ end the browser session by closing the MOVE tab and re-opening it in a new tab.

**Comment by candu** [comment]
This was fixed a while back by #796 , which added the `?login=1` GET parameter to successful login flow completions.
